sdk/src/Core/Amazon.Runtime/Internal/Auth/AWS4Signer.cs

```diff
@@ -499,7 +499,7 @@ public static string SetRequestBodyHash(IRequest request, string chunkedBodyHash
         public static string SetRequestBodyHash(IRequest request, bool signPayload, string chunkedBodyHash, int signatureLength)
         {
             // If unsigned payload, set the appropriate magic string in the header and return it
-            if (request.DisablePayloadSigning != null ? request.DisablePayloadSigning.Value : !signPayload)
+            if (request.UnsignedPayload || request.DisablePayloadSigning.GetValueOrDefault() || !signPayload)
```

Comment:
GetValueOrDefault will return `false` for the default value of DisablePayloadSigning. In that case it isn't part of the condition as it was before because before it would use the flag if it was `false` or `true` set explicitly and only use !signPayload if it was null. It didn't have to be `true`. Are you sure this logic is correct?

There is at least one test that sets DisablePayloadSigning to `false` explicitly and would still go into the if logic before.